core/src/services/seafile/backend.rs

```diff
@@ -0,0 +1,343 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use async_trait::async_trait;
+use http::StatusCode;
+use log::debug;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+use super::core::parse_dir_detail;
+use super::core::parse_file_detail;
+use super::core::SeafileCore;
+use super::error::parse_error;
+use super::lister::SeafileLister;
+use super::writer::SeafileWriter;
+use super::writer::SeafileWriters;
+use crate::raw::*;
+use crate::services::seafile::core::AuthInfo;
+use crate::*;
+
+/// Config for backblaze seafile services support.
+#[derive(Default, Deserialize)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct SeafileConfig {
+    /// root of this backend.
+    ///
+    /// All operations will happen under this root.
+    pub root: Option<String>,
+    /// server address of this backend.
+    ///
+    /// - If server is set, we will take user's input first.
+    /// - If not, we will try to load it from environment.
+    pub server: Option<String>,
+    /// username of this backend.
+    ///
+    /// - If username is set, we will take user's input first.
+    /// - If not, we will try to load it from environment.
+    pub username: Option<String>,
+    /// password of this backend.
+    ///
+    /// - If password is set, we will take user's input first.
+    /// - If not, we will try to load it from environment.
```

Comment:
Why env will we use?

Reply:
Are you suggesting that we shouldn't use environment variables as passwords?

Reply:
> Are you suggesting that we shouldn't use environment variables as passwords?

Sorry for my typo. I'm asking `What env will we use?`.

It looks we didn't read any seafile env in the builder.

Reply:
The description of the environment variable has been removed.